core/backend/src/test/sheetindex/SheetIndex.test.ts

```diff
@@ -0,0 +1,447 @@
+
+    const sheetIndex = SheetIndex.insert(iModel, modelId, "TestSheetIndex");
+    expect(Id64.isValidId64(sheetIndex)).to.be.true;
+  });
+
+  describe("Update", () => {
+    it("Priority", () => {
+      const subjectId = iModel.elements.getRootSubject().id;
+
+      const modelId = SheetIndexModel.insert(iModel, subjectId, "TestSheetIndexModel");
+      expect(Id64.isValidId64(modelId)).to.be.true;
+
+      const sheetIndex1Id = SheetIndex.insert(iModel, modelId, "TestSheetIndex");
+      expect(Id64.isValidId64(sheetIndex1Id)).to.be.true;
+
+      const folderId = SheetIndexFolder.insert(iModel, modelId, sheetIndex1Id, "TestFolder", 1);
+      expect(Id64.isValidId64(folderId)).to.be.true;
+
+      const folder = iModel.elements.tryGetElement<SheetIndexFolder>(folderId);
+      expect(folder).to.not.be.undefined;
+      expect(folder?.entryPriority).equals(1);
+
+      folder!.entryPriority = 0;
+      folder!.update();
+
+      const folderPostUpdate = iModel.elements.tryGetElement<SheetIndexFolder>(folderId);
+      expect(folderPostUpdate?.entryPriority).equals(0);
+    });
+
+    it("Parent", () => {
+      const subjectId = iModel.elements.getRootSubject().id;
+
+      const modelId = SheetIndexModel.insert(iModel, subjectId, "TestSheetIndexModel");
+      expect(Id64.isValidId64(modelId)).to.be.true;
+
+      const sheetIndex1Id = SheetIndex.insert(iModel, modelId, "TestSheetIndex-1");
+      expect(Id64.isValidId64(sheetIndex1Id)).to.be.true;
+
+      const sheetIndex2Id = SheetIndex.insert(iModel, modelId, "TestSheetIndex-2");
+      expect(Id64.isValidId64(sheetIndex2Id)).to.be.true;
+
+      const folderId = SheetIndexFolder.insert(iModel, modelId, sheetIndex1Id, "TestFolder", 1);
+      expect(Id64.isValidId64(folderId)).to.be.true;
+
+      const folder = iModel.elements.tryGetElement<SheetIndexFolder>(folderId);
+      expect(folder).to.not.be.undefined;
+
+      // Need to register class scheme?
+      // const relationship = iModel.relationships.tryGetInstance(SheetIndexOwnsEntries.classFullName, { sourceId: sheetIndexId, targetId: folderId });
+      const parentRel11 = iModel.relationships.tryGetInstanceProps(ElementOwnsChildElements.classFullName, { sourceId: sheetIndex1Id, targetId: folderId });
+      expect(parentRel11).to.not.be.undefined;
+
+      folder!.parent = new SheetIndexOwnsEntries(sheetIndex2Id);
+      folder!.update();
+
+      // Need to register class scheme?
+      // const relationship = iModel.relationships.tryGetInstance(SheetIndexOwnsEntries.classFullName, { sourceId: sheetIndexId, targetId: folderId });
+      const parentRel12 = iModel.relationships.tryGetInstanceProps(ElementOwnsChildElements.classFullName, { sourceId: sheetIndex1Id, targetId: folderId });
+      expect(parentRel12).to.be.undefined;
+
+      // Need to register class scheme?
+      // const relationship = iModel.relationships.tryGetInstance(SheetIndexOwnsEntries.classFullName, { sourceId: sheetIndexId, targetId: folderId });
```

Comment:
This isn't necessary, but is there a way to get an instance of a RelatedElement directly?  When I try to run this method I get the following error:

```
Bad Arg: class BisCore,ElementOwnsChildElements has no superclass
      at Function.generateClassForEntity (src\ClassRegistry.ts:103:13)
```
I assume this is because [NavigationRelationships are not Entities](https://github.com/iTwin/itwinjs-core/blob/c9d607822932cdc444205221c48f58fab3bab7d0/core/backend/src/NavigationRelationship.ts#L9).

I also found the [Relationship](https://www.itwinjs.org/reference/core-backend/relationships/relationship/) class.  Should `SheetIndexOwnsEntries` and `SheetIndexReferenceRefersToSheetIndex` be extending this class instead?

Reply:
More specifically, [ElementRefersToElement](https://www.itwinjs.org/reference/core-backend/relationships/elementreferstoelements/) seems like an appropriate base class for `SheetIndexReferenceRefersToSheetIndex`.

Reply:
Your assumption is correct: A Navigation-relationship is just backing-up a "foreign-key" column that is stored along an Element, so there's no separate "row" in the database for such relationship. 

On the other hand, a link-table relationship (like ElementRefersToElement subclasses) lead to separate rows stored in their own table, different from its source or target Element. 

Picking between the two during schema design primarily lays on questions about multiplicity (1-n vs. n-n) as well as data-ownership. 

It seems the tryGetInstance API only works with the latter kind because it returns an instance of "Relationship" - which expects it to be link-table. Maybe such API could be refactored to work more seamlessly with both kinds of relationships. ECSQL itself is cleaner on that regard... you could write a query based on any kind of relationship the same way (e.g. SELECT SourceECInstanceId, TargetECInstanceId FROM <any relationship - nav-property or link-table>).

So, finally answering your question, I don't think there's an API way to get to an instance of a RelatedElement directly, associated via a nav-property.

Reply:
Thanks for the informative answer.  I will leave the implementation as is, and will clean up these comments in the test file.